src/components/authentication-template.ts

```diff
@@ -0,0 +1,214 @@
+    preventDefault(event);
+
+    if (!trimString(this.password as string)) {
+      this.passwordField?.focus();
+      this.passwordError = " * password can't be empty";
+      return;
+    }
+
+    this.showLoadingIndicator = true;
+    const response = (await backendServiceHandler({
+      action: 'verify-password',
+      identifier: this.identifier,
+      password: trimString(this.password as string),
+    })) as ResponseModel;
+
+    if (response.success === true) {
+      // successfully authenticated
+      this.dispatchEvent(new Event('ia-authenticated'));
+    } else {
+      this.passwordField?.focus();
+      this.passwordError = ' * invalid password';
+    }
+
+    this.showLoadingIndicator = false;
+  }
+
+  /**
+   * set password value to property
+   *
+   * @param {Event} e
+   * @memberof AuthenticationTemplate
+   */
+  setPassword(e: Event) {
+    const input = e.target as HTMLInputElement;
+    this.password = input.value;
+    this.passwordError = '';
+  }
+
+  get iaPasswordVerification() {
+    return html`
+      <p>Please verify your password again to change protected settings.</p>
+      <div class="form-element">
+        <label for="password">Internet Archive password</label>
+        <input
+          type="password"
+          class="form-control password"
+          id="password"
+          name="password"
+          required
+          @input=${this.setPassword}
+        />
+        <img
+          class="password-icon"
+          src="https://archive.org/images/eye-crossed.svg"
+          @click=${(e: Event) =>
+            togglePassword(e, this.passwordField as HTMLInputElement)}
+          alt="Hide text"
+        />
+        <span class="error-field">${this.passwordError}</span>
+        <a
+          href="/account/forgot-password?email=${encodeURIComponent(
+            this.email
+          )}"
+          >Forgot password?</a
+        >
+      </div>
+      <div class="form-element">
+        <button
+          class="ia-button primary ${this.showLoadingIndicator
+            ? 'pointer-none'
+            : ''}"
+          @click=${(e: Event) => {
+            this.verifyIAPassword(e);
+          }}
+        >
+          ${this.showLoadingIndicator
+            ? html` <ia-activity-indicator
+                mode="processing"
+                class="go-button-loading-icon"
+              ></ia-activity-indicator>`
+            : 'Verify password'}
+        </button>
+      </div>
+    `;
+  }
+
+  get googleVerification() {
+    return html`
+      <p>Please sign in again to change protected settings.</p>
+      <div class="form-element footer">
+        <a @click=${() => (this.authenticationType = 'ia')}
+          >Prefer to use your Internet Archive password?</a
+        >
+      </div>
+      <slot></slot>
+    `;
+  }
+
+  render() {
+    return html`
+      <div class="authentication-template">
+        <form method="post" name="authentication-settings" autocomplete="off">
```

Comment:
q: do we need a `submit` callback that does all checks before sending?

Reply:
no we don't need here.